runtime-interface/src/lib.rs

```diff
@@ -148,6 +148,33 @@ unsafe fn sys_mprotect_interval(
     Err(MprotectError::OsError)
 }
 
+#[cfg(feature = "std")]
+fn mprotect_pages_vec(mem_addr: u64, pages: &[u32], protect: bool) -> Result<(), MprotectError> {
+    if pages.is_empty() {
+        return Ok(());
+    }
+
+    let mprotect = |start, count, protect: bool| unsafe {
+        let addr = mem_addr + (start * PageNumber::size()) as u64;
+        let size = count * PageNumber::size();
+        sys_mprotect_interval(addr, size, !protect, !protect, false)
+    };
+
+    // Collects continuous intervals of memory from lazy pages to protect them.
+    let mut start = pages[0] as usize; // Can't panic as we've checked `pages` is not empty
```

Comment:
Comments sideways are not looking good. Moreover `first` method could be used with the explanation:
```suggestion
    let mut start = pages.first().expect("checked `pages` is not empty") as usize;
```

Reply:
Done